src/components/uploadPicture/uploadImage.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect, useCallback } from "react";
+import styles from "./uploadImage.module.css";
+import { trpc } from "@/client/lib/trpc";
+import axios from "axios";
+import { FaCamera, FaCheckCircle } from "react-icons/fa";
+
+interface UploadImageComponentProps {
+  title: string;
+  fileName: string;
+}
```

Comment:
needs an `onUpload: (fileId: string) => void` handler as well so the parent component can receive uploaded files

Reply:
I added the onUpload as a prop as asked. I also called it later in the **uploadImageToCloud**  passing  the **Key**  from the finishFileUpload mutation after a successful upload. There has to be an onUpload handler when it is called in the parent component. Is this what is expected?

Reply:
almost, `onUpload()` should be called with the `file_id` returned by `finishFileUpload.mutateAsync()` instead of the intermediate `key`